modules/notes.js

```diff
@@ -0,0 +1,242 @@
+  team: 'T1FDR4NG3',
+  blocks: [ [Object] ],
+  thread_ts: '1603807314.106800'
+},
+...]
+ */
+
+  const categorizedData = {}
+  let project = ''
+  updates.forEach(({ text, user }) => {
+    const isActionDid = []
+    const isActionDoing = []
+    const isActionQuestions = []
+    const _input = text.split('\n')
+    _input.forEach((line, i) => {
+      if (line.toLowerCase().match(/^did/)) isActionDid.push(i)
+      if (line.toLowerCase().match(/^doing/)) isActionDoing.push(i)
+      if (line.toLowerCase().match(/^question/)) isActionQuestions.push(i)
+    })
+    const did = isActionDid.length ? _input.slice(isActionDid[0], isActionDoing[0]) : []
+    // const doing = isActionDoing.length && _input.slice(isActionDoing[0], isActionQuestions[0])
+    const questions = isActionQuestions.length ? _input.slice(isActionQuestions[0]) : []
+
+    const categorize = (input) => {
+      const [repo, subCat, _update] = parseSubject(input.replace(/^\W*\s*/, ''))
+      const update = `${_update} (<@${user}>)`
+
+      if (categorizedData[project][repo]) {
+        if (categorizedData[project][repo][subCat]) {
+          categorizedData[project][repo][subCat].push(update)
+        } else {
+          categorizedData[project][repo][subCat] = [update]
+        }
+      } else {
+        categorizedData[project][repo] = { [subCat]: [update] }
+      }
+    }
+
+    did.forEach((input) => {
+      // 'Did - Locus:'
+      if (input.toLowerCase().match(/^did/)) {
+        const { groups: { main_project } } = input.match(/(?<action>\w+)(\W*)(?<main_project>[A-Za-z]*)?/) || {}
+        project = main_project ? main_project.toLowerCase() : 'others'
+        categorizedData[project] = { ...categorizedData[project] }
+        // categorizedData: { locus: {}}
+        return
+      }
+      categorize(input)
+    })
+
+    questions.forEach((input) => {
+      if (input.toLowerCase().match(/^question/)) {
+        project = 'questions'
+        categorizedData[project] = { ...categorizedData[project] }
+        return
+      }
+      categorize(input)
+    })
+    /**
+     categorizedData = {
+      common: {
+        release: { others: [Array], nlp: [Array] },
+        avail: { others: [Array] }
+      },
+      locus: {
+        snoke: { builder: [Array], devops: [Array] },
+        hub: { others: [Array] },
+        firstorder: { builder: [Array] }
+      },
+      data: { data: { gam: [Array], others: [Array] } }
+    }
+     */
+  })
+  const _questions = categorizedData.questions
+  delete categorizedData.questions
+  let response = ''
+  Object.entries(categorizedData).forEach(([project, details]) => {
+    if (project !== 'others') response += `\n\n\n ${project.toUpperCase()} \n`
+    Object.entries(details).forEach(([repo, updates]) => {
+      response += `\n\n*_${repo}_* \n`
+      Object.entries(updates).forEach(([subCat, content]) => {
+        if (subCat === 'others') {
+          content.forEach((input) => (response += `    • ${input} \n`))
```

Comment:
would `\t` (tab) work better for readability of the code on intent of how much to indent?